src/openeo_gfmap/manager/job_splitters.py

```diff
@@ -0,0 +1,63 @@
+"""Job splitter functionalities, except input points/polygons to extract in the
+form of a GeoDataFrames.
+"""
+from typing import List
+
+import geopandas as gpd
+import h3
+
+
+def _resplit_group(polygons: gpd.GeoDataFrame, max_points: int) -> List[gpd.GeoDataFrame]:
+    """Performs re-splitting of a dataset of polygons in a list of datasets"""
+    for i in range(0, len(polygons), max_points):
+        yield polygons.iloc[i : i + max_points].reset_index(drop=True)
+
+
+def split_job_hex(
+    polygons: gpd.GeoDataFrame, max_points: int = 500, grid_resolution: int = 4
```

Comment:
Thought you mentioned `grid_resolution=3` to be the default? Need to sync this with @cbutsko. For WorldCereal it's likely that the splitting is already done in RDM, meaning that a sample already contains the gridbox ID of a specific resolution (currently 3) that allows easy querying of the RDM API to select only samples in one gridbox.
But for general functionality good to have it here too, we just might want to align the default resolution.